src/Cli/ConfigGenerator.cs

```diff
@@ -510,6 +510,83 @@ public static bool TryAddNewEntity(AddOptions options, RuntimeConfig initialRunt
             return true;
         }
 
+        /// <summary>
+        /// Tries to update the runtime settings based on the provided runtime options.
+        /// </summary>
+        /// <returns>True if the update was successful, false otherwise.</returns>
+        public static bool TryUpdateRuntimeSettings(UpdateRuntimeOptions options, FileSystemRuntimeConfigLoader loader, IFileSystem fileSystem)
+        {
+            if (!TryGetConfigFileBasedOnCliPrecedence(loader, options.Config, out string runtimeConfigFile))
+            {
+                return false;
+            }
+
+            if (!loader.TryLoadConfig(runtimeConfigFile, out RuntimeConfig? runtimeConfig))
+            {
+                _logger.LogError("Failed to read the config file: {runtimeConfigFile}.", runtimeConfigFile);
+                return false;
+            }
+
+            if (!TryUpdateDepthLimit(options, ref runtimeConfig))
+            {
+                return false;
+            }
+
+            return WriteRuntimeConfigToFile(runtimeConfigFile, runtimeConfig, fileSystem);
+        }
+
+        /// <summary>
+        /// Attempts to update the depth limit in the GraphQL runtime settings based on the provided value.
+        /// If a depth limit is provided in the options, it validates the limit to ensure it's an integer and within the valid range of 1 to Int32.MaxValue.
+        /// A depth limit of -1 is considered a special case that disables the GraphQL depth limit.
+        /// </summary>
+        /// <param name="options">Options including the new depth limit.</param>
+        /// <param name="runtimeConfig">Current config, updated if method succeeds.</param>
+        /// <returns>True if the update was successful, false otherwise.</returns>
+        private static bool TryUpdateDepthLimit(
+            UpdateRuntimeOptions options,
+            [NotNullWhen(true)] ref RuntimeConfig runtimeConfig)
+        {
+            if (options.DepthLimit is not null)
```

Comment:
we've started setting the standard that null values in the config means `default`. In the PR description it mentions that by default we have no depth limit. If someone had set a depth limit to some value, say 3, and then they want to go back to the default value, would their only option be to pass in the value that we have set as the default (currently int.MAX I presume).

Perhaps passing in null here should just go back to default behavior, and then if default behavior is a different value in the future, user doesnt have to know about this detail, just follow the pattern of null = default.

Reply:
if users wants to go back to the default value, i.e, remove current depth limit, they need use the command
`dab runtime --depth-limit -1`.

Reply:
> if users wants to go back to the default value, i.e, remove current depth limit, they need use the command `dab runtime --depth-limit -1`.

I see, so if in the future we have some default depth limit that is an actual value (not just removing any depth limit), -1 will still be used to go back to that default value?